### PR TITLE
Added verifier abstraction section

### PR DESCRIPTION
**Pull request by nedmsmith**
Added definition of verifier abstractions for describing verifier internals.

### CODE REVIEW

**Review by jdbeaney — APPROVED** [review]

File: draft-ietf-rats-corim.md [hunk 1]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+Reference Values Claims typically describe various possible states due to versioning,
+manufactruing practices, or supplier configuration options.
+* actual state: Claims that describe a Target Environment instance,
+given some constraining context, such as a point in time or a network connection.
+Endorsed Values and Evidence typically are Claims about actual state.
+* Group: The scope in which Evidence, Reference Values, Endorsed Values and Appraisal Policy
+are applied.
+An Attester may be composed of multiple components, where each component may
+represent a scope of appraisal.
+* Authority: Provenance linking a Claim to the entity that asserted it.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1323]: _(non-native speaker comment)_

I am not sure I understand "Provenance" in and by itself.  Maybe: "Provenance statement" or "Provenance assertion".

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+Typically, a Claim is asserted using a cryptographic key to digitally sign the Claim. A cryptographic key can be a proxy for a human or organizational entity.
+* Accepted Claims Set (ACS): A structure that holds EMT Claims that have been vetted
+following the appraisal process. The ACS describes the actual state of an Attesster that has been vetted by appraisal policy. The ACS keeps track of a Claim's authority.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1326]:
```suggestion
following the appraisal process. The ACS describes the actual state of an Attester that has been vetted by Appraisal Policy. The ACS also keeps track of a Claim's authority.
```

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+* Appraisal Policy: A description of the conditions, that if met, allow acceptance
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:1327]:
```suggestion
* Appraisal Policy: A description of the conditions that, if met, allow acceptance
```

**Review by thomas-fossati — APPROVED** [review]
Awesome and much needed stuff.
I have inlined a few editorial suggestions.

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
 For example, it is expected that the resources used during the initialisation
 phase can be amortised across multiple appraisals.
 
+## Verifier Abstraction
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1303]: Thank you for the text. I am wondering perhaps this paragraph fits more logically at start (i.e. Line 1284) before we start enumerating phases of Verification ?

File: draft-ietf-rats-corim.md [hunk 5]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+This document assumes Verifier implementations may differ.
+To facilitate description of normative Verifier behavior,
+this document uses abstract representation of Verifier internals.
+
+* Claim: A structure that has a key-value pairing.
+* Environment Measurement Tuple (EMT): A structure containing a set of environment
+Claims that describe a Target Environment and a set of measurement Claims that
+describe attributes of the Target Environment.
+* possible state: Claims that describe various alternative forms of a Target Environment.
+Reference Values Claims typically describe various possible states due to versioning,
```
Comment (yogeshbdeshpande) [inline draft-ietf-rats-corim.md:1314]:
```suggestion
Reference Values Claims are typically attributes of the Target Environment that describe various possible states due to versioning,
```

**Review by yogeshbdeshpande — APPROVED** [review]
I have left some comments for you to have a look and consider implementing them.

In general they are `non-blocking` hence I am approving the code review.

File: draft-ietf-rats-corim.md [hunk 6]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+manufactruing practices, or supplier configuration options.
+* actual state: Claims that describe a Target Environment instance,
+given some constraining context, such as a point in time or a network connection.
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1317]: Seems a bit too vague for my taste (allows for a lot of interpretation)

File: draft-ietf-rats-corim.md [hunk 7]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
 For example, it is expected that the resources used during the initialisation
 phase can be amortised across multiple appraisals.
 
+## Verifier Abstraction
+
+This document assumes Verifier implementations may differ.
+To facilitate description of normative Verifier behavior,
+this document uses abstract representation of Verifier internals.
+
+* Claim: A structure that has a key-value pairing.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1309]: Blocking issue: this must be RFC 9334 definitions where possible/feasible

File: draft-ietf-rats-corim.md [hunk 8]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+* Environment Measurement Tuple (EMT): A structure containing a set of environment
+Claims that describe a Target Environment and a set of measurement Claims that
+describe attributes of the Target Environment.
+* possible state: Claims that describe various alternative forms of a Target Environment.
```
Comment (henkbirkholz) [inline draft-ietf-rats-corim.md:1313]: Maybe should inherit `intended state` here? (there can be various allowed intended states)

File: draft-ietf-rats-corim.md [hunk 9]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+Reference Values Claims typically describe various possible states due to versioning,
+manufactruing practices, or supplier configuration options.
+* actual state: Claims that describe a Target Environment instance,
```
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1316]: Maybe current `operational state`?

Borrowing from https://www.rfc-editor.org/rfc/rfc8342.html#section-3 could possibly help here, in general.

File: draft-ietf-rats-corim.md [hunk 10]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+given some constraining context, such as a point in time or a network connection.
+Endorsed Values and Evidence typically are Claims about actual state.
+* Group: The scope in which Evidence, Reference Values, Endorsed Values and Appraisal Policy
+are applied.
```
Comment (andrew-draper) [inline draft-ietf-rats-corim.md:1320]:
```suggestion
* Group: A set of Evidence, Reference Values, Endorsed Values and Appraisal Policy which
are processed together.
```

File: draft-ietf-rats-corim.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1316]: In the editor's discussion today, incorporating terminology from https://www.ietf.org/archive/id/draft-dthaler-rats-endorsements-02.html was brought up as a prominent option. This requires a sync with Dave.

File: draft-ietf-rats-corim.md (hunk 10 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1320]: We can use this "Group definition" as a placeholder for now. Andrew noted that a better term is needed.

File: draft-ietf-rats-corim.md (hunk 7 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1309]: > A piece of asserted information, often in the form of a name/value pair. Claims make up the usual structure of Evidence and other RATS conceptual messages. Compare: claim [[RFC7519](https://datatracker.ietf.org/doc/html/rfc7519)].

File: draft-ietf-rats-corim.md [hunk 11]
```diff
@@ -1300,6 +1300,35 @@ they are not required to use the same internal data structures.
...
+Reference Values Claims typically describe various possible states due to versioning,
+manufactruing practices, or supplier configuration options.
+* actual state: Claims that describe a Target Environment instance,
+given some constraining context, such as a point in time or a network connection.
+Endorsed Values and Evidence typically are Claims about actual state.
+* Group: A set of Evidence, Reference Values, Endorsed Values and Appraisal Policies
+which are processed together.
+An Attester may be composed of multiple components, where each component may
+represent a scope of appraisal.
+* Authority: Provenance linking a Claim to the entity that asserted it.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1323]:
```suggestion
* Authority: The entity asserting that a claim is true.
```

**Review by nedmsmith — COMMENTED** [review]
multiple

File: draft-ietf-rats-corim.md (hunk 6 above)
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1317]:
```suggestion

```

File: draft-ietf-rats-corim.md (hunk 7 above)
Comment (andrew-draper) [inline draft-ietf-rats-corim.md:1309]:
```suggestion
* Claim: A key-value pairing encoding information.
```

File: draft-ietf-rats-corim.md (hunk 8 above)
Comment (henkbirkholz) [reply draft-ietf-rats-corim.md:1313]: See below: following Dave's wording for now.

File: draft-ietf-rats-corim.md (hunk 4 above)
Comment (nedmsmith) [reply draft-ietf-rats-corim.md:1303]: Either location works for me

File: draft-ietf-rats-corim.md [hunk 12]
```diff
@@ -1300,6 +1300,34 @@ they are not required to use the same internal data structures.
...
+
+This document assumes Verifier implementations may differ.
+To facilitate description of normative Verifier behavior,
+this document uses abstract representation of Verifier internals.
+
+* Claim: A piece of information, in the form of a key-value pair.
+* Environment Measurement Tuple (EMT): A structure containing a set of environment
+Claims that describe a Target Environment and a set of measurement Claims that
+describe attributes of the Target Environment.
+* possible state: Claims that describe various alternative forms of a Target Environment.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1313]:
```suggestion
* reference state: Claims that describe various alternative forms of a Target Environment.
```

File: draft-ietf-rats-corim.md [hunk 13]
```diff
@@ -1300,6 +1300,34 @@ they are not required to use the same internal data structures.
...
+
+This document assumes Verifier implementations may differ.
+To facilitate description of normative Verifier behavior,
+this document uses abstract representation of Verifier internals.
+
+* Claim: A piece of information, in the form of a key-value pair.
+* Environment Measurement Tuple (EMT): A structure containing a set of environment
+Claims that describe a Target Environment and a set of measurement Claims that
+describe attributes of the Target Environment.
+* reference state: Claims that describe various alternative forms of a Target Environment.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1313]:
```suggestion
* reference state: Claims that describe various alternative states of a Target Environment.
```

File: draft-ietf-rats-corim.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (nedmsmith) [reply draft-ietf-rats-corim.md]: draft-dthaler-rats-endorsements also uses 'point in time' in the definition. It uses "the "Attester" language which implies an entity instance. Maybe that is good enough here too?

[the remark this answers is not in the document]
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md]: > draft-dthaler-rats-endorsements also uses 'point in time' in the definition. It uses "the "Attester" language which implies an entity instance. Maybe that is good enough here too?

I'm happy aligning with draft-thaler.  I'd also be happy with something like:

> _actual state: Claims that describe a Target Environment instance at a given point in time._

File: draft-ietf-rats-corim.md [hunk 14]
```diff
@@ -1300,6 +1300,34 @@ they are not required to use the same internal data structures.
...
+this document uses abstract representation of Verifier internals.
+
+* Claim: A piece of information, in the form of a key-value pair.
+* Environment Measurement Tuple (EMT): A structure containing a set of environment
+Claims that describe a Target Environment and a set of measurement Claims that
+describe attributes of the Target Environment.
+* reference state: Claims that describe various alternative states of a Target Environment.
+Reference Values Claims typically describe various possible states due to versioning,
+manufactruing practices, or supplier configuration options.
+* actual state: Claims that describe a snapshot of the state of a Target Environment instance.
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:1316]:
```suggestion
* actual state: Claims that describe a Target Environment instance at a given point in time.
```

**Pull request merged**